openapi.yaml

```diff
@@ -1,279 +1,662 @@
-    email: isdataarchitecture@oregonstate.edu
-schemes:
-  - https
-host: api.oregonstate.edu
-basePath: /v1
-externalDocs:
-  description: GitHub Repository
-  url: https://github.com/osu-mist/express-api-skeleton
-produces:
-  - application/json
-consumes:
-  - application/json
+  version: "2.0.0"
+  title: Locations API
+  description: Get information about OSU locations.
+servers: 
+  - url: https://api.oregonstate.edu/v2
 security:
   - OAuth2:
-      - full
+    - full
 paths:
-  /pets:
+  /locations:
     get:
-      summary: List all pets
       tags:
-        - pets
-      description: Get all pets
-      operationId: getPets
+       - locations
+      summary: Get locations
+      description: "Gets all `Location` objects. Can be filtered by parameters."
+      operationId: getLocations
       parameters:
-        - $ref: '#/parameters/pageNumber'
-        - $ref: '#/parameters/pageSize'
         - in: query
-          name: species
-          description: Search by pet species
+          name: filter[query]
+          description: "Query to search through buildings based on building name and abbreviation"
           required: false
-          type: string
+          schema:
+            type: string
+        - in: query
+          name: filter[latitude]
+          description: "Latitude of the location. Should be used with lon parameter"
+          required: false
+          schema:
+            type: number
+            format: float
+        - in: query
+          name: filter[longitude]
+          description: "Longitude of the location. Should be used with lat parameter"
+          required: false
+          schema:
+            type: number
+            format: float
+        - in: query
+          name: filter[distance]
+          description: "The radius of the circle centred on the specified location. Default: 2"
+          required: false
+          schema:
+            type: number
+            format: float
+        - in: query
+          name: filter[distanceUnit]
+          description: "Distance can be specified in following units:
+            Mile - mi or miles (default);
+            Yard - yd or yards;
+            Feet - ft or feet;
+            Inch - in or inch;
+            Kilometer - km or kilometers;
+            Meter - m or meters;
+            Centimeter - cm or centimeters;
+            Millimeter - mm or millimeters;
+            Nautical mile - NM, nmi or nauticalmiles
+          "
+          required: false
+          schema:
+            type: string
+            enum: 
+              - mi
+              - miles
+              - yd
+              - yards
+              - ft
+              - feet
+              - in
+              - inch
+              - km
+              - kilometers
+              - m
+              - meters
+              - cm
+              - centimeters
+              - mm
+              - millimeters
+              - nmi
+              - NM
+              - nauticalmiles
```

Comment:
You can put them in a normal array since you have listed them in description. Example: https://github.com/osu-mist/persons-api/blob/v2/openapi.yaml#L118-L142